tests/integration_test.rs

```diff
@@ -0,0 +1,13 @@
+mod common;
```

Comment:
I'd avoid use of the term 'common' - it is by definition applicable to everything and therefore not a distinction.   It is probably suitable to add the common init function at the test root.   Actually in my mind this is on of those cases where duplication is not bad, so adding it to every test individually would be my preference.

Reply:
we can name it something else, but it has to be in it's own module, otherwise the cargo test runner will try to run the setup functions as a test each time

Maybe 'integration_util' or something similiar? Things like server setup and teardown will live here, along with the logging I put in there for now